src/lib/apiTypesV2/Entities.cpp

```diff
@@ -145,15 +145,29 @@ void Entities::fill(QueryContextResponse* qcrsP)
   for (unsigned int ix = 0; ix < qcrsP->contextElementResponseVector.size(); ++ix)
   {
     ContextElement* ceP = &qcrsP->contextElementResponseVector[ix]->contextElement;
-    Entity*         eP  = new Entity();
-
-    eP->id        = ceP->entityId.id;
-    eP->type      = ceP->entityId.type;
-    eP->isPattern = ceP->entityId.isPattern;
-    eP->creDate   = ceP->entityId.creDate;
-    eP->modDate   = ceP->entityId.modDate;
-
-    eP->attributeVector.fill(&ceP->contextAttributeVector);
-    vec.push_back(eP);
+    StatusCode* scP = &qcrsP->contextElementResponseVector[ix]->statusCode;
+
+    if (scP->code == SccReceiverInternalError)
+    {
+      // Do we need to release the memory allocated in 'vec' before returning? I don't
+      // think so, as the releasing logic in the upper layer will deal with that but
+      // let's do anyway just in case...
```

Comment:
No, this should not be necessary.
i would remove the call to `vec.release` and the comment and let valgrind tell us (it *is* a special casa as it gives back an error).
Unnecessary code and comments only confuse ...